src/hotspot/share/gc/serial/cardTableRS.hpp

```diff
@@ -41,10 +40,27 @@ class CardTableRS : public CardTable {
   friend class VMStructs;
   // Below are private classes used in impl.
   friend class VerifyCTSpaceClosure;
-  friend class ClearNoncleanCardWrapper;
 
   void verify_space(Space* s, HeapWord* gen_start);
 
+  static bool is_dirty(const CardValue* const v) {
+    return !is_clean(v);
+  }
+
+  static bool is_clean(const CardValue* const v) {
+    return *v == clean_card_val();
+  }
+
+  static void clear_cards(CardValue* start, CardValue* end);
+
+  static CardValue* find_first_dirty_card(CardValue* const start_card,
+                                          CardValue* const end_card);
+  template<typename Func>
+  CardValue* find_first_clean_card(CardValue* const start_card,
+                                   CardValue* const end_card,
+                                   CardTableRS* ct,
+                                   Func&& object_start);
```

Comment:
Neither rvalue references nor universal references are permitted in HotSpot code (yet).  Usual style is to
take function arguments by value.  The copy may not even matter (may be elided/optimized away), given
this function is called once and potentially inlined.  Maybe declare this function inline to encourage that?
I'm waffling over rejecting this, since I think we ought to at least permit universal references, but that
hasn't been proposed yet.